jupyter/muehle-gui.ipynb

```diff
@@ -257,36 +455,259 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "def update_gui(state):\n",
-    "    # TODO: ADD INTERACTIVITY\n",
-    "    canvas[1].clear()\n",
-    "    ((stash_p1, stash_p2), squares) = state\n",
-    "    \n",
-    "    # update pieces on the board\n",
-    "    for i in range(len(squares)):\n",
-    "        for j in range(len(squares[i])):\n",
-    "            draw_stone(canvas[1], coords[i][j], squares[i][j])\n",
+    "%run ./muehle-game.ipynb"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### getClickedStone\n",
+    "(TODO: Erklärung)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def getClickedStone(x, y):\n",
+    "    for value in av:\n",
+    "        if value - PLAYER_PIECE_RADIUS <= x <= value + PLAYER_PIECE_RADIUS:\n",
+    "            logger.info(f'value for x = {x} found: {value}')\n",
+    "            x = value\n",
+    "        if value - PLAYER_PIECE_RADIUS <= y <= value + PLAYER_PIECE_RADIUS:\n",
+    "            logger.info(f'value for y = {y} found: {value}')\n",
+    "            y = value\n",
+    "    stone = None\n",
+    "    for i in range(len(coords)):\n",
+    "        if stone is not None:\n",
+    "            break\n",
+    "        for j in range(len(coords[i])):\n",
+    "            if coords[i][j] == (x,y):\n",
+    "                stone = (i, j)\n",
     "\n",
-    "    # update pieces on the stash\n",
+    "    logger.info(f'stone clicked: {stone}')\n",
+    "    return stone"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### convertStateTupleToList\n",
+    "(TODO: Erklärung)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def convertStateTupleToList(s):\n",
+    "    return [list(s[0]), [list(s[1][0]), list(s[1][1]), list(s[1][2])]]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### convertStateListToTuple\n",
+    "(TODO: Erklärung)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def convertStateListToTuple(s):\n",
+    "    return (tuple(s[0]), (tuple(s[1][0]), tuple(s[1][1]), tuple(s[1][2])))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Klasse GameState\n",
+    "\n",
+    "(TODO: Erklärung)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class GameState:\n",
+    "    def __init__(self, state = None):\n",
+    "        if state is None:\n",
+    "            state = ((9, 9), (\n",
+    "            (' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '),\n",
+    "            (' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '),\n",
+    "            (' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ')\n",
```

Comment:
Den State `s0` sollten wir in der Zukunft auslagern in die `game-utils`

Reply:
ich habe den Zustand jetzt ganz oben in game-utils definiert als `START_STATE`.
Eventuell kann er aber auch in die gui zu den anderen Konstanten, weil er ja nur da gebraucht wird?

Reply:
✔️